indico/util/user.py

```diff
@@ -88,6 +97,17 @@ def principal_from_fossil(fossil, allow_pending=False, allow_groups=True, allow_
         if group.group is None and not allow_missing_groups:
             raise ValueError('Multipass group does not exist: {}:{}'.format(provider, id_))
         return group
+    elif category:
+        category_roles = list(
+            itertools.chain.from_iterable(
+                c.roles for c in category.chain_query.options(joinedload('roles'))
+            )
+        )
+        role = CategoryRole.get(id_)
```

Comment:
This can also be simplified like this:

    role = CategoryRole.query.filter_by(id=id_).join(category.chain_query.subquery()).first()

This could also go into another `CategoryRole` staticmethod (`get_for_category(cat, id)`).